### PR TITLE
Add new domain status pages navigation

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

Adds a separate component to handle the domain status navigation links. This is initial PR that will allow us to split the work in smaller PRs. We've decided to update the navigation now with the new component without feature-flagging it.

Here's how the navigation looks:

registered domain
Screenshot 2020-03-25 at 15 10 41

mapped domain
Screenshot 2020-03-25 at 15 10 48

wpcom domain
Screenshot 2020-03-25 at 15 10 55

#### Testing instructions

* Check that all links in domain navigation are working
* Check that we're recording events for the wpcom domain links

### CODE REVIEW

**Comment by hambai** [comment]
Known issue I'm working on is the wpcom domain tracking is not working right now

File: client/my-sites/domains/domain-management/edit/navigation/index.jsx [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+		if ( ! this.isDomainInNormalState() && ! this.isDomainInGracePeriod() ) {
+			return null;
+		}
+
+		return (
+			<DomainManagementNavigationItem
+				path={ domainManagementContactsPrivacy( selectedSite.slug, domain.name ) }
+				materialIcon="chrome_reader_mode"
+				text={ translate( 'Contacts and privacy' ) }
+				description={ null && translate( 'Privacy protection: on' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Privacy protection' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-status-new-navigation)

File: client/my-sites/domains/domain-management/edit/navigation/index.jsx [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+			return;
+		}
+
+		const path = domainManagementDomainConnectMapping( selectedSite.slug, domain.name );
+
+		return (
+			<DomainManagementNavigationItem
+				path={ path }
+				materialIcon="cached"
+				text={ translate( 'Connect your domain' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Connect Your Domain' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-status-new-navigation)

**Comment by hambai** [comment]
Cool - I'll update the PR to clean up the old navigation then

File: client/my-sites/domains/domain-management/edit/navigation/index.jsx [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+		const { domain, isLoadingPurchase, purchase, selectedSite, translate } = this.props;
+		const domainType = domain && domain.type;
+
+		if ( ! domain.currentUserCanManage ) {
+			return null;
+		}
+
+		const title =
+			domainTypes.MAPPED === domainType
+				? translate( 'Delete domain mapping' )
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **34** times:
`translate( 'Delete Domain Mapping' )` *ES Score: 11*

File: client/my-sites/domains/domain-management/edit/navigation/index.jsx [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+				: translate( 'Delete your domain permanently' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/navigation/index.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **4** times:
`translate( 'Delete Your Domain Permanently' )` *ES Score: 11*

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3118261

Thank you @hambai for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3118261).

**Pull request merged**